core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -356,11 +359,34 @@ private void setFinalResult(
               TimeUnit.NANOSECONDS);
         }
       }
+      // log the warnings if they have NOT been disabled
+      if (LOG.isWarnEnabled()
+          && !executionInfo.getWarnings().isEmpty()
+          && !executionProfile.getBoolean(DefaultDriverOption.REQUEST_DISABLE_LOG_WARNINGS)) {
+        logServerWarnings(executionInfo.getWarnings());
+      }
     } catch (Throwable error) {
       setFinalError(error, callback.node, -1);
     }
   }
 
+  private void logServerWarnings(List<String> warnings) {
+    // use the RequestLogFormatter to format the query
+    StringBuilder statementString = new StringBuilder();
+    requestLogFormatter.appendRequest(
+        statement,
+        executionProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_QUERY_LENGTH, 500),
+        executionProfile.getBoolean(DefaultDriverOption.REQUEST_LOGGER_VALUES, false),
+        executionProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUES, 0),
+        executionProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUE_LENGTH, 0),
+        statementString);
```

Comment:
👍 for re-reading all options every time, we want them to be modifiable at runtime.